airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -154,17 +158,22 @@ def _calculate_cursor_datetime_from_state(self, stream_state: Mapping[str, Any])
             return self.parse_date(stream_state[self.cursor_field.eval(self.config)])
         return datetime.datetime.min.replace(tzinfo=datetime.timezone.utc)
 
-    def _format_datetime(self, dt: datetime.datetime):
+    def _format_datetime(self, dt: datetime.datetime) -> str:
```

Comment:
Nit: looks like this is no longer used so can be deleted.

Reply:
This is not a nit since this is what is sent to the API and therefore, it should be used. I'll update this. My guess is that when we save the end of the slice, we should parse and format using the first `cursor_datetime_format`